lib/segment/src/problems/unindexed_field.rs

```diff
@@ -0,0 +1,330 @@
+use std::any::TypeId;
+use std::collections::{HashMap, HashSet};
+use std::sync::OnceLock;
+use std::time::Duration;
+
+use http::{HeaderMap, Method, Uri};
+use issues::{Action, Code, ImmediateSolution, Issue, Solution};
+use itertools::Itertools;
+
+use crate::common::operation_error::OperationError;
+use crate::data_types::text_index::{TextIndexParams, TextIndexType, TokenizerType};
+use crate::json_path::{JsonPathInterface, JsonPathV2};
+use crate::types::{
+    AnyVariants, Condition, FieldCondition, Filter, Match, MatchValue, PayloadFieldSchema,
+    PayloadKeyType, PayloadSchemaParams, PayloadSchemaType,
+};
+
+#[derive(Debug)]
+pub struct UnindexedField {
+    field_name: JsonPathV2,
+    field_schemas: HashSet<PayloadFieldSchema>,
+    collection_name: String,
+    endpoint: Uri,
+    instance_id: String,
+}
+
+impl UnindexedField {
+    pub fn slow_search_threshold() -> Duration {
+        static SLOW_SEARCH_THRESHOLD: OnceLock<Duration> = OnceLock::new();
+
+        *SLOW_SEARCH_THRESHOLD.get_or_init(|| {
+            Duration::from_millis(
+                std::env::var("QDRANT_SLOW_SEARCH_THRESHOLD")
+                    .ok()
+                    .and_then(|var| var.parse::<u64>().ok())
+                    .unwrap_or(300),
+            )
+        })
```

Comment:
A nitpick: I think we should use seconds here instead, to prevent using different units everywhere. We can do `0.3` as a float.

What do you think?

Reply:
I think it is a good suggestion, changed